bridge/scripts/lib/madnetFactoryTasks.ts

```diff
@@ -213,45 +413,84 @@ task("deployCreate", "deploys a contract from the factory using create")
     }
   });
 
-task("upgradeDeployedProxy", "deploys a contract from the factory using create")
+task(DEPLOY_PROXY, "deploys a proxy from the factory")
+  .addParam(
+    "salt",
+    "salt used to specify logicContract and proxy address calculation"
+  )
+  .addOptionalParam("factoryAddress")
+  .setAction(async (taskArgs, hre) => {
+    const network = hre.network.name;
+    const factoryAddress = await getFactoryAddress(network, taskArgs);
+    const factoryBase = await hre.ethers.getContractFactory(MADNET_FACTORY);
+    const factory = factoryBase.attach(factoryAddress);
+    const txResponse = await factory.deployProxy(taskArgs.salt);
+    const receipt = await txResponse.wait();
+    const proxyAddr = getEventVar(receipt, DEPLOYED_PROXY, CONTRACT_ADDR);
+    const proxyData: ProxyData = {
+      proxyAddress: proxyAddr,
+      salt: taskArgs.salt,
+      factoryAddress,
+      gas: receipt.gasUsed,
+      receipt,
+    };
+    return proxyData;
+  });
+
+task(UPGRADE_DEPLOYED_PROXY, "deploys a contract from the factory using create")
   .addParam("contractName", "logic contract name")
-  .addParam("factoryName", "Name of the factory contract")
-  .addParam("factoryAddress", "factory deploying the contract")
   .addParam(
     "logicAddress",
     "address of the new logic contract to upgrade the proxy to"
   )
+  .addOptionalParam("factoryAddress", "factory deploying the contract")
   .addOptionalParam(
     "initCallData",
-    "data used to initialize initializable contracts"
+    "input initCallData args in a string list, eg: --initCallData 'arg1, arg2'"
+  )
+  .addOptionalParam(
+    "outputFolder",
+    "output folder path to save deployment arg template and list"
   )
   .setAction(async (taskArgs, hre) => {
-    let factoryAddress = await getFactoryAddress(taskArgs);
-    let MadnetFactory = await hre.ethers.getContractFactory(MADNET_FACTORY);
-    //grab the salt from the logic contract
-    let Salt = await getBytes32Salt(taskArgs.contractName, hre);
-    //get logic contract interface
-    let logicContract: any = await hre.ethers.getContractFactory(
+    const network = hre.network.name;
+    const factoryAddress = await getFactoryAddress(network, taskArgs);
```

Comment:
Suggestions: When receiving a factory address, check if that address is from a factory before proceeding with the rest of the script. Send a good error message in case the address is not to a contract factory.